sidecar/config.go

```diff
@@ -604,3 +668,74 @@ rm -rf /root/backup
 	}
 	return nil
 }
+
+func (cfg *Config) executeCloneRestore() error {
+	// Check directory exist, create if not exist.
+	if _, err := os.Stat(utils.DataVolumeMountPath); os.IsNotExist(err) {
+		os.Mkdir(utils.DataVolumeMountPath, 0755)
+	}
+
+	// Empty the directory.
+	dir, err := ioutil.ReadDir(utils.DataVolumeMountPath)
+	if err != nil {
+		return fmt.Errorf("failed to read datadir %s", err)
+	}
+	for _, d := range dir {
+		os.RemoveAll(path.Join([]string{utils.DataVolumeMountPath, d.Name()}...))
+	}
+	// Xtrabackup prepare and apply-log-only.
+	cmd := exec.Command(xtrabackupCommand, "--defaults-file="+utils.ConfVolumeMountPath+"/my.cnf", "--use-memory=3072M", "--prepare", "--apply-log-only", "--target-dir=/backup/"+cfg.XRestoreFrom)
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to xtrabackup prepare apply-log-only : %s", err)
+	}
+	// Xtrabackup Prepare.
+	cmd = exec.Command(xtrabackupCommand, "--defaults-file="+utils.ConfVolumeMountPath+"/my.cnf", "--use-memory=3072M", "--prepare", "--target-dir=/backup/"+cfg.XRestoreFrom)
+	cmd.Stderr = os.Stderr
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to xtrabackup prepare : %s", err)
+	}
+	// Get the backup binlong info.
+	gtid, err := GetXtrabackupGTIDPurged("/backup/" + cfg.XRestoreFrom)
+	if err == nil {
```

Comment:
Throw the error when return error?

Reply:
Cannot throw error， donot has gtid， is a normal condition.

Reply:
Add the handle about the err.

Reply:
It's normal .